cmake/TestImpactFramework/LYTestImpactFramework.cmake

```diff
@@ -439,6 +452,31 @@ function(ly_test_impact_write_config_file CONFIG_TEMPLATE_FILE BIN_DIR)
     message(DEBUG "Test impact framework post steps complete")
 endfunction()
 
+#! ly_test_impact_write_pytest_file: writes out the test information utilised by our TIAF testing tools, using the data derived from the build generation process.
+# 
+function(ly_test_impact_write_pytest_file)
+
+    # For each configuration type, compile the build info we need and add it to our array
+    set(build_configs "")
+    foreach(config_type ${LY_CONFIGURATION_TYPES})
+        set(config "${LY_TEST_IMPACT_WORKING_DIR}/${config_type}/${LY_TEST_IMPACT_PERSISTENT_DIR}/${LY_TEST_IMPACT_CONFIG_FILE_NAME}")
+        ly_file_read("cmake/TestImpactFramework/LyTestImpactBuildConfigEntry.in" build_config)
```

Comment:
I think we can get rid of this extra template and compile the JSON array of dicts ourselves, something like:

```
"build_configurations": [
    {
        "build_config": "debug",
        "confg_file": "path/to/config/file.json",
        "report_file": "path/to/report.json"
    },
    {
        "build_config": "profile",
        "confg_file": "path/to/config/file.json",
        "report_file": "path/to/report.json"
    },
    {
        "build_config": "release",
        "confg_file": "path/to/config/file.json",
        "report_file": "path/to/report.json"
    }
]
```

Have a look at `ly_test_impact_write_test_enumeration_file` for an example of compiling the JSON in this way.

Reply:
Done